webapp/app/.server/service/signatures.ts

```diff
@@ -152,9 +227,21 @@ export async function saveEmergencySignature(
       }
       break;
     }
-    case "ExecuteEmergencyUpgradeZKFoundation":
-      // TODO: Check how to validate this signature.
+    case "ExecuteEmergencyUpgradeZKFoundation": {
+      const isValid = await isValidSignatureZkFoundation(
+        await zkFoundationAddress(),
+        signature,
+        await emergencyBoardAddress(),
```

Comment:
Can we parallelize this?

Reply:
We agreed on leaving this for next release.